packages/react-native-windows-init/src/Cli.ts

```diff
@@ -171,22 +171,38 @@ function getLatestMatchingVersion(
         },
       );
     } else {
-      // Assume that versionSemVer is actually a tag
-      npm.packages.release(
-        pkg,
-        versionSemVer,
-        (err: any, details: {version: string}[]) => {
-          if (err) {
-            reject(err);
-          } else if (details && details.length > 0) {
-            resolve(details[0].version);
-            return;
-          }
-          reject(
-            new Error(`No matching version of ${pkg}@${versionSemVer} found`),
-          );
-        },
-      );
+      try {
+        exec(
+          `npm info ${pkg}@${versionSemVer} version --json`,
+          (err, stdout, _stderr) => {
+            try {
+              if (!err) {
+                let candidates = JSON.parse(stdout);
+                if (typeof candidates === 'string') {
+                  resolve(candidates);
+                  return;
+                }
+                candidates = candidates.sort(semver.rcompare);
+                if (candidates && candidates.length > 0) {
```

Comment:
can there be more than one match (e.g. wildcards)? if so, print them and error out